src/simularium/FrameRecorder.ts

```diff
@@ -0,0 +1,143 @@
+import { ArrayBufferTarget, Muxer } from "mp4-muxer";
+
+/**
+ * Records frames to an MP4 file using the WebCodecs API.
+ * Resulting file is passed to the front end implementation
+ * for download/use, via the handleFile() callback.
+ *
+ *
+ * Note that the VideoCodecs API is unavailable in some browsers, including Firefox,
+ * as of 2/6/2024. Viewport will not call these methods on firefox.
+ *
+ */
+export class FrameRecorder {
+    private getCanvas: () => HTMLCanvasElement | null;
+    private handleFile: (videoBlob: Blob) => void;
+    private encoder: VideoEncoder | null;
+    private muxer?: Muxer<ArrayBufferTarget>;
+    public isRecording: boolean;
+    private frameIndex: number;
+    public supportedBrowser: boolean;
+
+    constructor(
+        getCanvas: () => HTMLCanvasElement | null,
+        handleFile: (videoBlob: Blob) => void
+    ) {
+        this.getCanvas = getCanvas;
+        this.handleFile = handleFile;
+        this.encoder = null;
+        this.isRecording = false;
+        this.frameIndex = 0;
+        this.supportedBrowser = !/firefox|fxios/i.test(navigator.userAgent);
+    }
```

Comment:
was originally going to handle all browser checking in the website/front end, but @meganrm suggested we check for firefox further down and calling `stop()` on firefox was leading to errors. you could circumvent this on the front end, but this prevent it in the first place.

Reply:
I wonder if there is a better way to init `supportedBrowser` by actually checking for the api that will be used?

Reply:
We do the check in Colorizer like this: 
```
  public static isSupported(): boolean {
    // Check if WebCodecs API exists
    return typeof VideoEncoder === "function";
  }
```

https://github.com/allen-cell-animated/nucmorph-colorizer/blob/main/src/components/Export.tsx#L326

Reply:
Oh interesting @ShrimpCryptid Just came back here to reply to Dan's comment because
 `this.supportedBrowser = "VideoEncoder" in window;` 
 seemed to be working. I think they are more or less equivalent?